### ISSUE TITLE
Silent push notifications stop to be delivered in background (after a while)

### ISSUE THREAD

**Issue by skauting** [firebase/quickstart-ios#258]
Hi, 

my application receives silent push notifications... works fine if the app is in foreground, but when moved to background it only receives them for a while (8-10 minutes) and after nothing for about one hour and after again few received (for 8-10 minutes) and again nothing...

I also realised, that calling
`        FIRMessaging.messaging().appDidReceiveMessage(userInfo)
`
in background is causing this warning:

>         <FIRMessaging/WARNING> FIRMessaging receiving notification in invalid state 2

I am wondering, if these two things can't be linked... Firebase stops sending silent push messages after a while, because they are not confirmed when received in background??? 

Thanks a lot, Jan

**Comment by rsattar** [comment]
Hi @skauting! The `FIRMessaging.messaging().appDidReceiveMessage(userInfo)` call is not linked with whether or not silent messages are deliverable. That warning (should be improved) is whether or not the notification can be sent to Analytics. 

If you're able to receive silent messages while the app is in the background it's going through Apple's APNs servers, as a silent notification. Apple also states that [silent notifications are _not_ guaranteed delivery, and is on a "best effort" basis](https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/CreatingtheNotificationPayload.html#//apple_ref/doc/uid/TP40008194-CH10-SW1). So it's possible that the APNs is no longer delivering your silent notifications. Also, if you force quit an app (by swiping it away in the task switcher), silent notifications are not delivered by Apple to your app until the next app restart.

Hope that helps!

**Comment by skauting** [comment]
Hi @rsattar, thanks a lot. I continued to search and it seems, my issues is not linked to Firebase, but to silent notification management is general. I found similar questions at stackoverflow, facing same problems and not using Firebase at all. More details here, if interested.

http://stackoverflow.com/questions/43672785/firebase-ios-silent-push-notification-stop-to-be-delivered-in-background-after

Thanks a lot, closing this issues here.
